### PR TITLE
[SPARK-20027][DOCS] Compilation fix in java docs.

### PR DESCRIPTION
**Pull request by ScrapCodes**
## What changes were proposed in this pull request?

During build/sbt publish-local, build breaks due to javadocs errors. This patch fixes those errors.

## How was this patch tested?

Tested by running the sbt build.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Hi @ScrapCodes, I just happened to look at this PR and ran `jeykill build` against this PR.

It seems there are few more instances to make javadoc8 working as below:

```
[error] .../spark/mllib/target/java/org/apache/spark/ml/fpm/FPGrowthParams.java:17: error: bad use of '>'
[error]    * Number of partitions (>=1) used by parallel FP-growth. By default the param is not set, and
[error]                            ^
[error] .../spark/sql/core/src/main/java/org/apache/spark/api/java/function/FlatMapGroupsWithStateFunction.java:30: error: reference not found
[error]  * {@link org.apache.spark.sql.KeyValueGroupedDataset#flatMapGroupsWithState(
[error]           ^
[error] .../spark/core/target/java/org/apache/spark/TaskContextImpl.java:10: error: reference not found
[error]  * {@link TaskMetrics} &amp; {@link MetricsSystem} objects are not thread safe.
[error]           ^
[error] .../spark/core/target/java/org/apache/spark/TaskContextImpl.java:10: error: reference not found
[error]  * {@link TaskMetrics} &amp; {@link MetricsSystem} objects are not thread safe.
[error]
```

Should we maybe sweep these together if I found them correctly?

File: common/network-common/src/main/java/org/apache/spark/network/crypto/ClientChallenge.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 /**
  * The client challenge message, used to initiate authentication.
  *
- * @see README.md
+ * see README.md
```
Comment (srowen) [inline common/network-common/src/main/java/org/apache/spark/network/crypto/ClientChallenge.java]: This should just be a sentence rather than fragment.

**Review by srowen — APPROVED** [review]

**Comment by ScrapCodes** [comment]
Hi @HyukjinKwon !, Thank for looking at this. I did one pass through all those errors, they seem to be inside the generated code. Please notice `/target` in the path.

File: common/network-common/src/main/java/org/apache/spark/network/crypto/ClientChallenge.java (hunk 1 above)
Comment (ScrapCodes) [inline common/network-common/src/main/java/org/apache/spark/network/crypto/ClientChallenge.java]: Thanks !

**Comment by HyukjinKwon** [comment]
Yea, It generates java codes for Java API. I suggested to fix these together because they look similar instances and I didn't want to open another minor PR for this (BTW, [FlatMapGroupsWithStateFunction.java](https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/api/java/function/FlatMapGroupsWithStateFunction.java) seems not a generated code).

If you think this should be separately handled in another PR, then I am happy to do so.

If you meant it is not an issue,

I tried both

```
 rm -fr target
 ./build/sbt clean
 ./build/sbt unidoc
```

and 

`cd docs && jekyll build`

I see 

```
[error] (spark/javaunidoc:doc) javadoc returned nonzero exit code
```

and

```
jekyll 3.3.1 | Error:  Unidoc generation failed
```

due to those four errors and I don't see `_site` and the Java API documentation after `jekyll build` as written in [docs#generating-the-documentation-html](https://github.com/apache/spark/tree/master/docs#generating-the-documentation-html).

**Comment by HyukjinKwon** [comment]
If my suggestion is the reason to hold off this PR currently and you are not sure of it, I am fine. I can double-check and deal with this in a separate PR.

**Comment by ScrapCodes** [comment]
@HyukjinKwon, Fixing those errors require fixing unidoc plugin's java code generation. IMO, that has a broader scope. What do you think ?

**Comment by HyukjinKwon** [comment]
Sorry for making this PR dragged. I am fine. Let me cope with it separately.

**Comment by ScrapCodes** [comment]
Actually thank you for your comments, I missed one file in swarm of errors from generated files.

**Comment by ScrapCodes** [comment]
Jenkins, retest this please.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**